### ISSUE TITLE
Multiple columns with same "database column"

### ISSUE THREAD

**Issue by gabgren** [rappasoft/laravel-livewire-tables#597]
Hi, 

I have a table where some columns use the same source to display different values.

Example:
Column 1: Order ID
Column 2: Customer (i need to figure out the customer for this order with some logic based on the Order ID)
Column 3: Created on (based on a value in a json string stored in the 'dates' database column)
Column 4: Due on (based on a value in the same json string stored in the 'dates' database column)

It works, except for Sorting (and probably filtering)

The columns builder looks like this:
```
public function columns(): array
    {
        return [
             Column::make('ID', 'id'),
             Column::make('Customer', 'id')
                 ->format(function ($value) {
                     $Order = Order::find($value);
                     $Customer = $Customer::find($Order->customer_id);
                     return $Customer->name;
                }),
             Column::make('Created', 'dates')
                 ->format(function ($value) {
                     $dates = json_decode($value);
                     return $dates->created_on;
                }),
             Column::make('Due Date', 'dates')
                 ->format(function ($value) {
                     $dates = json_decode($value);
                     return $dates->due_on;
                }),
```

Let's imagine that the 'dates' database column for order 1234 contains this string:
```
{
"created_on":"2021-01-01",
"due_on":"2021-01-05"
}
```

My code works perfectly and gives me something like:
```
ID           Customer          Created          Due Date
1234       John Smith       2021-01-01         2021-01-05
```

BUT! Since columns 3 and 4 share the same input (dates), I cannot sort them. If I try to sort by "Due Date", it gets sorted by "Created" date and the filter added is called "Created".

Am I missing something? Or did I just hacked my way around this limitation by having more than one row with the same input ?
Or maybe there is another way of doing what I am doing?

I also wonder if, apart from the Sort/Filter bug that it causes, this can have negative effect on something else...

**Comment by rappasoft** [comment]
You hacked your way around a limitation. I'm not sure there's a way around it. Though you can try to specifically select the same column twice with different aliases, and then use raw order bys on each column?

**Comment by gabgren** [comment]
Hi ! thanks for the quick response :)
I've played a little bit more with it and yeah, I understand that this is not how it's meant to be. I ended up using custom row view for the display. I guess what I was looking for was a custom filter or custom sort function. 

For example, we might need to sort by Customer name (A-Z), where the 'customer' column is just an int (id of the customer). Maybe that would be a feature to add? 

I saw that you're currently working on V2 :) Keep up the good work! This is great !
Cheers,
Gab
